src/xhydro/frequency_analysis/uncertainities.py

```diff
@@ -192,14 +192,20 @@ def calc_q_iter(
         The return periods to calculate quantiles for.
     small_regions_threshold : int, optional
         The threshold for removing small regions. Default is 5.
+    l1 : xr.DataArray, optional
+        First L-moment (location) values. L-moment can be specified for ungauged catchments.
+        If `None`, values are taken from ds_moments_iter.
 
     Returns
     -------
     xarray.DataArray
         Quantiles for each bootstrap sample and group.
     """
-    # We select groups for only one id
-    ds_temp = ds_groups[[var]].sel(id=bv).dropna("group_id", how="all")
+    # We select groups for all or one id
+    if id == "all":
```

Comment:
```suggestion
    if bv == "all":
```

This wasn't raising any error since `id` is a built-in function!